tutorials/02_intermediate/gridspec.py

```diff
@@ -1,59 +1,162 @@
-        specifies the geometry of the grid that a subplot will be
+        Specifies the geometry of the grid that a subplot will be
         placed. The number of rows and number of columns of the grid
         need to be set. Optionally, the subplot layout parameters
         (e.g., left, right, etc.) can be tuned.
 
     :class:`~matplotlib.gridspec.SubplotSpec`
-        specifies the location of the subplot in the given *GridSpec*.
+        Specifies the location of the subplot in the given *GridSpec*.
 
     :func:`~matplotlib.pyplot.subplot2grid`
-        a helper function that is similar to :func:`~matplotlib.pyplot.subplot`
+        A helper function that is similar to :func:`~matplotlib.pyplot.subplot`,
         but uses 0-based indexing and let subplot to occupy multiple cells.
+
+    :func:`~matplotlib.pyplot.subplots`
+        perhaps the primary function used to create figures and axes.
+        It's also similar to :func:`~matplotlib.pyplot.subplot`,
+        but creates and places all axes on the figure at once.
+
 """
+
 import matplotlib.pyplot as plt
 import matplotlib.gridspec as gridspec
 
+############################################################################
+# Basic Quickstart Guide
+# ======================
+#
+# These first two examples show how to create a basic 4-by-4 grid using
+# both :func:`~matplotlib.pyplot.subplots` and :mod:`~matplotlib.gridspec`.
+#
+# Using :func:`~matplotlib.pyplot.subplots` is quite simple.
+# It returns a :class:`~matplotlib.figure.Figure` instance and an array of
+# :class:`~matplotlib.axes.Axes` objects.
+
+fig1, f1_axes = plt.subplots(ncols=2, nrows=2)
+fig1.tight_layout()
+
+############################################################################
+# For such a simple use case, :mod:`~matplotlib.gridspec` is perhaps overly
+# verbose.
+# You have to create the figure and :class:`~matplotlib.gridspec.GridSpec`
+# instance separately, then pass the elements gridspec instance to the
+# :func:`~matplotlib.figure.Figure.add_subplot` method to create the axes
+# objects.
+# The elements of the gridspec are accessed as if they where elements of
+# a numpy array of the same shape.
+
+fig2 = plt.figure()
+spec2 = gridspec.GridSpec(ncols=2, nrows=2)
+f2_ax1 = fig2.add_subplot(spec2[0, 0])
+f2_ax2 = fig2.add_subplot(spec2[0, 1])
+f2_ax3 = fig2.add_subplot(spec2[1, 0])
+f2_ax4 = fig2.add_subplot(spec2[1, 1])
+fig2.tight_layout()
+
+#############################################################################
+# When you want to have plots of different sizes, however,
+# :mod:`~matplotlib.gridspec` becomes indispensable and provides a couple
+# of options.
+# The method shown here initializes a uniform grid specification,
+# and then uses typical numpy indexing and slices to allocate multiple
+# "cells" for a given subplot.
+
+fig3 = plt.figure()
+spec3 = gridspec.GridSpec(ncols=3, nrows=3)
+anno_opts = dict(xy=(0.5, 0.5), xycoords='axes fraction',
```

Comment:
and maybe just add a comment to remind people that `add_subplot` returns an `axes` object so you're chaining them together with `annotate`